src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Reliability/SqlConfigurableRetryLogicLoader.cs

```diff
@@ -273,5 +298,26 @@ private static void OnRetryingEvent(object sender, SqlRetryingEventArgs args)
             SqlClientEventSource.Log.TryTraceEvent("{0}, Last exception:<{1}>", msg, lastException.Message);
             SqlClientEventSource.Log.TryAdvancedTraceEvent("<ADV>{0}, Last exception: {1}", msg, lastException);
         }
+
+        private static ICollection<int> SplitErrorNumberList(string list)
+        {
+            if (!string.IsNullOrEmpty(list))
+            {
+                string[] parts = list.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
+                if (parts != null && parts.Length > 0)
+                {
+                    HashSet<int> set = new HashSet<int>();
+                    for (int index = 0; index < parts.Length; index++)
+                    {
+                        if (int.TryParse(parts[index], System.Globalization.NumberStyles.AllowLeadingWhite | System.Globalization.NumberStyles.AllowTrailingWhite, null, out int value))
+                        {
+                            set.Add(value);
+                        }
+                    }
+                    return set;
+                }
+            }
+            return null;
```

Comment:
`.Select(x => Convert.ToInt32(x)).ToList()` would return an empty collection instead of null right?

Reply:
True, and in this case the null wouldn't have failed but would cause the default set to be used which would mean  you could no longer set an empty list. Changed to return an empty hashset.